src/main/java/nl/opengeogroep/safetymaps/server/stripes/FotoFunctionActionBean.java

```diff
@@ -0,0 +1,200 @@
+import net.sourceforge.stripes.action.UrlBinding;
+import net.sourceforge.stripes.validation.Validate;
+import nl.opengeogroep.safetymaps.server.db.Cfg;
+import nl.opengeogroep.safetymaps.server.db.DB;
+import static nl.opengeogroep.safetymaps.server.db.JSONUtils.rowToJson;
+import org.apache.commons.dbutils.QueryRunner;
+import org.apache.commons.dbutils.handlers.MapListHandler;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+/**
+ *
+ * @author martijn
+ */
+@StrictBinding
+@MultipartConfig
+@UrlBinding("/api/foto")
+public class FotoFunctionActionBean implements ActionBean {
+
+    private static final Log log = LogFactory.getLog(FotoFunctionActionBean.class);
+    private static final String TABLE = "\"FotoFunctie\"";
+    private ActionBeanContext context;
+    private String PATH = "";
+
+    @Validate
+    private FileBean picture;
+
+    @Validate
+    private String fileName;
+
+    @Validate
+    private String extraInfo;
+
+    @Validate
+    private String voertuigNummer;
+
+    @Validate
+    private String incidentNummer;
+
+    @Validate
+    private String type;
+
+    public String getVoertuigNummer() {
+        return voertuigNummer;
+    }
+
+    public void setVoertuigNummer(String voertuigNummer) {
+        this.voertuigNummer = voertuigNummer;
+    }
+
+    public String getIncidentNummer() {
+        return incidentNummer;
+    }
+
+    public void setIncidentNummer(String incidentNummer) {
+        this.incidentNummer = incidentNummer;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getFileName() {
+        return fileName;
+    }
+
+    public void setFileName(String fileName) {
+        this.fileName = fileName;
+    }
+
+    public String getExtraInfo() {
+        return extraInfo;
+    }
+
+    public void setExtraInfo(String extraInfo) {
+        this.extraInfo = extraInfo;
+    }
+
+    @Override
+    public void setContext(ActionBeanContext context) {
+        this.context = context;
+    }
+
+    @Override
+    public ActionBeanContext getContext() {
+        return context;
+    }
+
+    public FileBean getPicture() {
+        return picture;
+    }
+
+    public void setPicture(FileBean picture) {
+        this.picture = picture;
+    }
+
+    @DefaultHandler
+    public Resolution foto() throws IOException, ServletException {
+        JSONObject response = new JSONObject();
+        try {
+            PATH = Cfg.getSetting("fotofunctie");
+
+            response.put("result", false);
+            if (extraInfo == null) {
+                extraInfo = "";
+            }
+
+            if (incidentNummer == null) {
+                incidentNummer = "N.V.T.";
+            }
+
+            insertIntoDb();
+            final File file = new File(PATH + fileName);
```

Comment:
Security vulnerability: fileName is not sanitized by checking for file separators